dpctl/_sycl_core.pxd

```diff
@@ -63,3 +63,4 @@ cdef class SyclQueue:
     cpdef SyclContext get_sycl_context (self)
     cpdef SyclDevice get_sycl_device (self)
     cdef DPPLSyclQueueRef get_queue_ref (self)
+    cpdef memcpy (self, dest, src, int count)
```

Comment:
Did you mean `cdef` ?

Reply:
For `c` part of `cpdef` to be efficient, `dest` and `src` need have type declarations. Looks like they can only be `Memory` objects. 

The method does not use `self` at all, so should it be a static method of the class?

Reply:
The method uses `self` in code `DPPLQueue_memcpy(self.queue_ptr, c_dest, c_src, count)`.

Reply:
Perhaps it would be better to not expose `memcpy` as a stand-alone symbol, but rather make it two method `copyto` and `copyfrom` of `Memory` object. That way it could copy to/from another Memory object, or host memory of any object that supports a buffer protocol.